xbmc/addons/addoninfo/AddonInfoBuilder.cpp

```diff
@@ -360,8 +367,26 @@ bool CAddonInfoBuilder::ParseXML(const AddonInfoPtr& addon, const TiXmlElement*
       if (element && element->GetText() != nullptr)
         addon->m_packageSize = StringUtils::ToUint64(element->GetText(), 0);
 
-      /* Parse addon.xml "<news lang="..">...</news>" */
+      /* Parse addon.xml "<news lang="..">...</news>"
+       *
+       * In the event that the changes log (news) in addon.xml is empty, check
+       * whether it is an installed addon and read a changelog.txt as a
+       * replacement, if available. */
       GetTextList(child, "news", addon->m_changelog);
+      if (addon->m_changelog.empty() && !isRepoXMLContent && !addonPath.empty())
+      {
+        using XFILE::auto_buffer;
+        using XFILE::CFile;
+
+        const std::string changelog = URIUtils::AddFileToFolder(addonPath, "changelog.txt");
+        if (CFile::Exists(changelog))
+        {
+          CFile file;
+          auto_buffer buf;
+          if (file.LoadFile(changelog, buf) > 0)
+            addon->m_changelog["en_GB"].assign(buf.get(), buf.size());
```

Comment:
can we get the `en_GB` string (as the default language) from somewhere? Or define it beforehand on an anonymous namespace?

Reply:
My English is on strike, what do you mean exactly?
Allow other languages there and make them available by means of a function?

E.g.:
```
addon->m_changelog[GetLanguage()].assign(buf.get(), buf.size());
```

Wouldn't make sense then, because no translations are possible there on text file.

Reply:
I mean simply to define the en_GB string somewhere and reuse it as KODI_DEFAULT_LANGUAGE or something similar. Someone reading the code (and not familiar with our defaults) probably can't figure out why we are only setting the British English value. Another option is probably to get it from some central place (so that if we change defaults it's reflected everywhere)

Reply:
Have added a new commit about where change all related places to it.